internal/services/securitycenter/security_center_server_vulnerability_assessment_hybrid_vm_resource.go

```diff
@@ -0,0 +1,146 @@
+		Read:   resourceServerVulnerabilityAssessmentHybridVMRead,
+		Delete: resourceServerVulnerabilityAssessmentHybridVMDelete,
+
+		// TODO: replace this with an importer which validates the ID during import
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(5 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"hybrid_machine_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.HybridMachineID,
+			},
+		},
+	}
+}
+
+func resourceServerVulnerabilityAssessmentHybridVMCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	hybridMachineId, err := computeParse.HybridMachineID(d.Get("hybrid_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	vulnerabilityAssessment, err := client.Get(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(vulnerabilityAssessment.Response) {
+			return fmt.Errorf("checking for presence of existing Advanced Threat Protection for %s: %+v", *hybridMachineId, err)
+		}
+	}
+
+	if vulnerabilityAssessment.ID != nil && *vulnerabilityAssessment.ID != "" {
+		return tf.ImportAsExistsError("azurerm_security_center_server_vulnerability_assessment_hybrid_vm", *vulnerabilityAssessment.ID)
+	}
+
+	vulnerabilityAssessment, err = client.CreateOrUpdate(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
+	if err != nil {
+		return fmt.Errorf("create Server Vulnerability Assessment for %s: %+v", *hybridMachineId, err)
+	}
+
+	if vulnerabilityAssessment.ID != nil {
+		d.SetId(*vulnerabilityAssessment.ID)
+	}
+
+	// Take some time for resource to be visible after creating. Hence the retry logic
+	readError := resourceServerVulnerabilityAssessmentRead(d, meta)
+	for retry := 1; d.Id() == "" && retry < 11; retry++ {
+		time.Sleep(time.Duration(retry) * time.Second)
+		readError = resourceServerVulnerabilityAssessmentRead(d, meta)
+	}
+	return readError
+}
+
+func resourceServerVulnerabilityAssessmentHybridVMRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	hybridMachineId, err := computeParse.HybridMachineID(d.Get("hybrid_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
```

Comment:
(similar to above) can we use the HybridVM Resource ID and then parse `d.ID()` instead?

We can't guarantee in the Read function that the schema fields are available, so there's a crash here